core/src/main/java/io/smallrye/openapi/api/models/media/SchemaImpl.java

```diff
@@ -82,61 +131,79 @@ public static void addTypeObserver(Schema observable, Schema observer) {
             obs.typeObservers = ModelUtil.add(observer, obs.typeObservers, ArrayList<Schema>::new);
         }
 
-        observer.setType(observable.getType());
+        setTypeRetainingNull(observer, observable.getType());
     }
 
     public static SchemaImpl copyOf(Schema other) {
-        SchemaImpl clone = (SchemaImpl) MergeUtil.mergeObjects(new SchemaImpl(), other);
-        clone.required = copy(clone.required, () -> new ArrayList<>(clone.required));
-        clone.enumeration = copy(clone.enumeration, () -> new ArrayList<>(clone.enumeration));
-        clone.items = copy(clone.items, () -> copyOf(clone.items));
-
-        clone.allOf = copy(clone.allOf, () -> clone.allOf
-                .stream()
-                .map(SchemaImpl::copyOf)
-                .collect(Collectors.toList()));
-
-        clone.properties = copy(clone.properties, () -> {
-            Map<String, Schema> copiedProperties = new LinkedHashMap<>(clone.properties.size());
-            clone.properties.forEach((k, v) -> copiedProperties.put(k, copyOf(v)));
-            return copiedProperties;
-        });
-
-        clone.additionalPropertiesSchema = copy(clone.additionalPropertiesSchema,
-                () -> copyOf(clone.additionalPropertiesSchema));
-
-        clone.xml = copy(clone.xml, () -> MergeUtil.mergeObjects(new XMLImpl(), clone.xml));
-        clone.externalDocs = copy(clone.externalDocs,
-                () -> MergeUtil.mergeObjects(new ExternalDocumentationImpl(), clone.externalDocs));
-
-        clone.oneOf = copy(clone.oneOf, () -> clone.oneOf
-                .stream()
-                .map(SchemaImpl::copyOf)
-                .collect(Collectors.toList()));
+        if (other == null) {
+            return new SchemaImpl();
+        }
```

Comment:
The logic of `copyOf` is very similar to `MapBasedModelImpl.mergeFrom`, we may be able to remove one of them.

Reply:
It's slightly different - where `from` has a property but `to` does not, the value from `from` is used directly. When the method returns, both `from` and `to` reference the same object. This matches the behaviour of `MergeUtil.mergeObjects` and means `mergeFrom` is only really suitable if you're going to throw both of the source objects away and just use the result object.

For this reason I recommend keeping both, even though the logic is similar.